### PR TITLE
Only discard the files changes inside the lane

### PR DESCRIPTION
**Pull request by estib-vega**
### Problem
When using the file context menu inside a given lane to discard the changes, all changes to that file across all lanes are discarded.
This is either weird or unexpected behavior.

### Solution
Only discard the changes inside the lane that the context menu was opened from.

### Issues
This fixes issue https://github.com/gitbutlerapp/gitbutler/issues/4137

### CODE REVIEW

File: crates/gitbutler-branch-actions/src/virtual.rs [hunk 1]
```diff
@@ -183,28 +178,44 @@ pub fn unapply_ownership(
...
+                .iter()
+                .filter(|o| files.contains(&o.file_path.to_string_lossy().to_string()))
+                .cloned()
+                .collect::<Vec<gitbutler_branch::OwnershipClaim>>();
+
+            BranchOwnershipClaims {
+                claims: filtered_ownership_claims,
+            }
+        })
+        .context("failed to determin ownership to unapply")?;
```
Comment (mtsgrd) [inline crates/gitbutler-branch-actions/src/virtual.rs]: typo!

Comment (estib-vega) [reply crates/gitbutler-branch-actions/src/virtual.rs]: Oops!
Fixed :)

File: apps/desktop/src/lib/vbranches/branchController.ts [hunk 2]
```diff
@@ -226,10 +226,11 @@ export class BranchController {
 		}
 	}
 
-	async unapplyFiles(files: LocalFile[]) {
+	async unapplyFiles(branchId: string, files: LocalFile[]) {
 		try {
 			await invoke<void>('reset_files', {
 				projectId: this.projectId,
+				branchId,
 				files: files.flatMap((f) => f.path).join('\n')
```
Comment (Byron) [inline apps/desktop/src/lib/vbranches/branchController.ts]: Can't this be a list of paths right away? My edits now assume this works, and I hope you can validate that.

**Review by Byron — COMMENTED** [review]
Thanks so much for all your recent work here (*making missing projects deletable was great!*).

I took the liberty to edit the rust-side of things so that it's idiomatic and less wasteful. However, I also tried something new which is to pass `Vec<PathBuf>` from the frontend to the backend directly. In the frontend, this means that a string-array should be passed and I think it should convert to `Vec<PathBuf>` automatically. However, I didn't try it as I didn't adjust the frontend code.

It would be great if it could work like this, but if not, `Vec<String>` will be fine, and a conversion to `Vec<PathBuf>` in the backend.

Thanks again!

**Comment by estib-vega** [comment]
Hey @Byron! Your improvements are more than welcome haha.
I try to adjust the FE code accordingly and give it a go.

Thanks!

**Comment by mtsgrd** [comment]
I added an e2e testing label here as a reminder this might be a good thing to cover.

**Review by ndom91 — APPROVED** [review]
LGTM :partying_face:

**Pull request merged**